doc/designs/adtrust/samba-domain-member.md

```diff
@@ -0,0 +1,429 @@
+for resources accessible via SMB. SMB protocol identity model is based on a
+Windows NT concept of security identifiers (SIDs) and access control lists
+(ACLs) which is not directly compatible with a concept of identities employed
+in POSIX environment model. Thus, Samba suite does serve as a translation layer
+between the two environments.
+
+Active Directory is an extension of Windows NT identity model where identity
+information is stored in a database exposed to the world via a combination of
+LDAP and SMB protocols, with authentication provided with both password
+(NTLMSSP) and Kerberos methods. Systems in Active Directory are organized into
+logical groups, domains, where some nodes, domain controllers, are used to
+store domain-specific information and others, domain members, are utilizing the
+information via SMB, LDAP, and Kerberos protocols.
+
+SMB protocol has a mechanism for encapsulating and channeling through itself
+other types of requests, expressed as an access to "files" over a specialized
+share `IPC$`. There are multiple interfaces provided by a typical domain
+controller and domain member servers, most well-known ones are LSA (local
+security authority, documented in [MS-LSAD] and [MS-LSAT]) and NETLOGON remote
+protocol (documented in [MS-NRPC]). LSA remote procedure calls are used, among
+other needs, for retrieving identity information about SIDs and their
+relationship to other objects. NETLOGON, as its name suggests, is utilized for
+authentication in a domain environment, across domains, and across forests of
+domains.
+
+In a traditional domain member set up, the member machine has no possession of
+a particular user credentials. Instead, it relies on its own connection to its
+own domain controller to identify a user and to proxy a user's authentication
+to the domain controller of the domain a user belongs to. In case a user is
+performing a remote authentication using Kerberos, a remote system has to
+present a Kerberos ticket to the domain member's SMB service, like with any
+other Kerberos services.
+
+To operate as a domain member in a FreeIPA domain, thus, Samba needs a FreeIPA
+master to be configured as a domain controller and a FreeIPA client needs to be
+configured in a specific way to allow Samba to talk to a domain controller.
+This document overviews a set of implementation tasks to achieve the domain
+member operation. Most of these tasks are related to FreeIPA components but
+some of changes required do belong to Samba itself.
+
+## Domain member configuration overview
+
+Samba suite, when running as a domain member, starts two daemons:
+
+- `smbd`, the main process which handles network connections, file system
+  operations, and remote procedure calls like LSA and NETLOGON. Each connection
+is handled by a separate `smbd` child;
+
+- `winbindd`, a process to perform identity resolution for all configured and
+  known domains. Each domain is handled by a separate `winbindd` child.
```

Comment:
Only domains which have an active connection are handled by a child in the meantime. This means you have only a child for your own domain running. There could be another child for a trusted domain if idmap_ad is configured. Normally all trusted domain users auth's are sent to your own domain.